install_config/upgrading/manual_upgrades.adoc

```diff
@@ -1272,3 +1285,147 @@ common issues:
+----
+subscription-manager register --consumerid=6699375b-06db-48c4-941e-689efd6ce3aa
+----
+
+[IMPORTANT]
+====
+When the deployment has been declared successful it's important that the old
+host be unregistered with `subscription-manager clean` to avoid creating a
+situation where the environment is out of compliance.
+====
+
+endif::openshift-enterprise[]
+
+=== Preparing for an upgrade
+
+link:#upgrading-masters[After the Master and etcd servers have been upgraded]
+you will need to ensure that your current production Nodes as either blue or
+green.  In this example the currently installation environment will be blue and
+the new environment will be green.
+
+----
+oc label --all nodes color=blue
+----
+
+There's quite a bit of flexibility here.  Perhaps there was only some subset of
+Nodes that required the uptime guarantees of a blue-green deployment.  In that
+case the `-l` flag can be used to match a subset of the environment using a
+selector.
+
+Next link:../install/advanced_install.html#adding-nodes-advanced[the green
+environment] should be created for any Nodes that are to be replaced.  The
+`color=green` label can be applied by Ansible using the `openshift_node_labels`
+variable for each Node.  The most critical attribute of the Node at this step
+is that the `openshift_schedulable=false` variable be set.  The goal is for no
+workloads to be scheduled until these Nodes are determined to be healthy.
+
+Once the green Nodes are in `Ready` state they can be made scheduleable. The
+blue Nodes are then disabled so that no new Pods are run on them.
+
+----
+oadm manage-node --schedulable=true --selector=color=green
+oadm manage-node --schedulable=false --selector=color=blue
+----
+
+A common practice is to scale the registry and router Pods until they are
+migrated to the green Nodes.  For these Pods a canary deployment approach is
+commonly used.  Scaling them up will make them immediately active on the new
+Nodes.  Pointing the deployment configs to the new image will cause a rolling
+update to happen.  However, because of Node anti-affinity and the fact that the
+blue Nodes are still unschedulable the deployments to the old Nodes will fail.
+At this point the registry and router deployments can be scaled down to the
+original number of Pods.  At any given point the original number of Pods is
+still available so no capacity is lost.
+
+=== Warming the new Nodes
+
+In order for Pods to be migrated from the blue environment to the green the
+images must be pulled.  Network latency and and load on the registry can cause
+delays if there isn't sufficient capacity built in to the environment. Often
+the best way to minimize impact to the running system is trigger new Pod
+deployments that will land on the new Nodes.  This can be accomplished by
+importing new ImageStreams.
+
+Quite often a major release of OpenShift is the motivation for a blue-green
+deployment.  In that case there will be new ImageStreams available for users of
```

Comment:
Consider rewording:
When this occurs, there are no new ImageStreams available...